middleware_test.go

```
@@ -0,0 +1,101 @@
+		if r.Header.Get("X-Mock-Header") != "Hello" {
+			t.Errorf("Expected X-Mock-Header: 'Hello', got '%s'", r.Header.Get("X-Mock-Header"))
+		}
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	mw := &mockMiddleware{}
+
+	// Instantiate client with middleware options
+	client, err := New(server.URL[7:], &Options{
+		Creds:        credentials.NewStaticV4("mockkey", "mocksecret", ""),
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofumpt)